src/app.test.js

```diff
@@ -40,16 +41,36 @@ const addTodo = (todoList, text, completed) => {
   }
 };
 
-test("renders empty todo list when starts", () => {
-  // setup
+const setUser = (userName = "Attila") => {
+  const userInput = screen.getByPlaceholderText("Please enter username");
+  const setUserButton = screen.getByText("Set User");
+
+  userEvent.type(userInput, userName);
+  userEvent.click(setUserButton);
+};
+
+test("does not allow to enter / add new todo until user is not set", () => {
+  // act
   const todoList = screen.getByTestId("todo-list");
+  addTodo(todoList, "Take wife for a walk!");
```

Comment:
how does "adding" a todo item help when input / button is disabled?

Reply:
Wont help in this instance as we don't check if this will appear in the list....was used in the previous assertion